js/main.ts

```diff
@@ -24,12 +25,14 @@ export default function denoMain() {
   libdeno.builtinModules["deno"] = deno;
   Object.freeze(libdeno.builtinModules);
 
+  version.deno = startResMsg.denoVersion();
```

Comment:
I would consider adding a set function that is exported from `version.ts` and marked `@internal` to do the setting, like we have with [`setEnv()` now](https://github.com/denoland/deno/blob/master/js/os.ts#L136).  This will make it more maintainable in the long term and keep logic out of here.  And `setVersions()` should freeze the `version` object too.